test/WebJobs.Script.Tests/ExtensionBundle/ExtensionBundleManagerTests.cs

```diff
@@ -407,22 +401,120 @@ public void LimitMaxVersion(string versionRange, string hostConfigVersion, strin
                 }
             }
 
-            var resolvedVersion = ExtensionBundleManager.FindBestVersionMatch(range, new List<string>()
-            { "3.7.0", "3.10.0", "3.11.0", "3.15.0", "3.14.0", "2.16.0", "3.13.0", "3.12.0", "3.9.1", "2.12.1", "2.18.0", "3.16.0", "2.19.0", "3.17.0", "4.0.2", "2.20.0", "3.18.0", "4.1.0", "4.2.0", "2.21.0", "3.19.0", "3.19.2", "4.3.0", "3.20.0" },
-            ScriptConstants.DefaultExtensionBundleId, hostingConfiguration);
+            var options = GetTestExtensionBundleOptions(BundleId, versionRange);
+            var manager = GetExtensionBundleManager(options, GetTestAppServiceEnvironment());
+
+            var resolvedVersion = manager.FindBestVersionMatch(range, GetLargeVersionsList(), ScriptConstants.DefaultExtensionBundleId, hostingConfiguration);
+
+            Assert.Equal(expectedVersion, resolvedVersion);
+        }
+
+        [Theory]
+        [InlineData(ScriptConstants.LatestPlatformChannelName, "[4.*, 5.0.0)", "4.2.0", "4.2.0")]
+        [InlineData(ScriptConstants.StandardPlatformChannelName, "[4.*, 5.0.0)", "4.2.0", "4.2.0")]
+        [InlineData(ScriptConstants.ExtendedPlatformChannelName, "[4.*, 5.0.0)", "4.2.0", "4.2.0")]
```

Comment:
I'm not sure I understand why these don't result in 4.1.0?

Reply:
The list of available bundles in this test (highest to lowest version) is: [ 4.3.0, 4.2.0, 4.1.0, ... 3.9.1].

The platform channel rules result in 4.2.0 (because latest is 4.3.0 and next is 4.2.0), and the max is also 4.2.0. It resolves to that value. 

There are cases below that have a max of 4.3.0, but resolves to 4.2.0 b/c the channel resolved to a lower value. I'll add a test for the reverse (max is 4.1.0, channel resolves to 4.2.0, so it should be 4.1.0).